gridstatus/isone_api/isone_api.py

```diff
@@ -65,7 +66,10 @@ def __init__(
         self.initial_delay = min(sleep_seconds, 60.0)
         self.max_retries = min(max(0, max_retries), 10)
 
-    # TODO(kladar) abstract this out to a base class since it is shared with ERCOT API logic
+    def parse_problematic_datetime(self, date_string: str) -> datetime:
+        dt = datetime.strptime(date_string, "%Y-%m-%dT%H:%M:%S.%f%z")
```

Comment:
When running `ISONEAPI().get_hourly_load_forecast('2024-03-11')`, I get the error `TypeError: strptime() argument 1 must be str, not Timestamp` which originates from this function.

Reply:
Mmm, yeah I seem to get both errors actually (dt accessor and string parse), probably because I'm trying to do this operation on the whole column series, even though certain datetimes want one and others want another method. I think it is the actual DST fall back/spring forward intervals don't get recognized as datetimes, and then the other intervals are timestamps and thus fail the string parsing. One second.

![CleanShot 2024-10-18 at 13 15 56@2x](https://github.com/user-attachments/assets/e8b6b018-03c2-4c6a-b8d9-00634f780438)

Reply:
Just confirming that I get the .dt error when attempting to format with 

```
df["BeginDate"] = pd.to_datetime(df["BeginDate"], format="%Y-%m-%dT%H:%M:%S.%f%z")
df["CreationDate"] = pd.to_datetime(df["CreationDate"], format="%Y-%m-%dT%H:%M:%S.%f%z")
```

![CleanShot 2024-10-18 at 13 24 02@2x](https://github.com/user-attachments/assets/2fdd019b-2fb3-4711-9c16-1328eb7cb870)

But a quick instance check on the problematic seems to fix it. I could just parse all the datetimes like that instead of trying the pd.to_datetime at all, but kind of want to avoid the parse to string and then call strptime wherever possible, only using it on the intervals/days that need it. 

I'm not sure that we are actually missing intervals in the database, but we must be because of the strptime failure, but my dupe SQL isn't picking it up.

Reply:
Okay, I tried the new logic with both DST start and DST end and it seems to correctly parse the datetimes. Maybe we should add tests for these dates? (We can do in a follow-up PR).